tests/ApplicationTest.php

```diff
@@ -24,4 +27,26 @@ public function runs()
 
         $this->assertTrue($called);
     }
+
+    /**
+     * @test
+     */
+    public function register_app_instance_to_container()
+    {
+        $app = Bridge::create();
+
+        $instance = null;
+
+        $app->get('/', function (App $app, ResponseInterface $response) use(&$instance) {
+            $instance = $app;
+            return $response;
+        });
+
+        $app->handle(RequestFactory::create());
+
+        $this->assertInstanceOf(
+            App::class,
+            $instance
+        );
```

Comment:
This could be replaced with:

```php
$this->assertSame($app, $instance);
```

That is a stricter assertion. That way we check that it's exactly the same instance of the object, and not a new `App` instance that PHP-DI could have created on the fly (for any reason).

Reply:
By the way feel free to add a new commit to the PR (i.e. same branch), don't worry about rebasing if you don't know that.

Reply:
I just replaced `assertInstanceOf` with `assertSame`